src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -23,34 +23,45 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comments = requestToString(request);
-    float commentsScore = calculateSentimentScore(comments);
+    Float commentsScore = calculateSentimentScore(comments);
     videoAnalysis = new VideoAnalysis(commentsScore);
     String json = new Gson().toJson(videoAnalysis);
 
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
-  // Converts JSON String of a request (that is in the form of an array) to a String, separated by
-  // periods. If the Json cannot be converted to an array, an empty String is returned. 
-  // Ex. "["a","b","c"]" --> "a. b. c"
-  public String requestToString(HttpServletRequest request) throws IOException {
+  /**
+   * Converts Json String of a request (that is in the form of an array) to a String, separated by
+   * periods. If the Json cannot be converted to an array or is an empty array, null is returned. 
+   * Ex. "["a","b","c"]" --> "a. b. c"
+   */
+  private String requestToString(HttpServletRequest request) throws IOException {
     Gson gson = new Gson();
     
     String textStr = request.getReader().lines().collect(Collectors.joining());
-    List<String> textList;
+
+    if (textStr.equals("[]")) {
+      return null;
+    }
+
     try {
-      textList = Arrays.asList(gson.fromJson(textStr, String[].class));
+      List<String> textList = Arrays.asList(gson.fromJson(textStr, String[].class));
+      return String.join(". ", textList);
     } catch (JsonSyntaxException e) {
       System.out.println(e.getMessage());
-      textList = Arrays.asList();
+      return null;
     }
-
-    return String.join(". ", textList);
   }
 
-  // Calculates sentiment score of text. The score is from -1 (negative) to +1 (positive).
-  public float calculateSentimentScore(String text) throws IOException {
+  /**
+   * Calculates sentiment score of text. The score is from -1 (negative) to +1 (positive).
+   */ 
+  private Float calculateSentimentScore(String text) throws IOException {
+    if (text == null) {
+      return null;
+    }
+
     Document doc =
         Document.newBuilder().setContent(text).setType(Document.Type.PLAIN_TEXT).build();
     LanguageServiceClient languageService = LanguageServiceClient.create();
```

Comment:
Can something go wrong in the language service? If so, how to they tell you that something went wrong? How should handle this? The front end might be interested to know if something in stack failed. Would recommend returning an HTTP error explaining that the language service failed for a reason not related to the request in this servlet.